diff/diff_test.go

```diff
@@ -91,6 +94,268 @@ func TestParseHunksAndPrintHunks(t *testing.T) {
 	}
 }
 
+func TestParseFileDiffHeaders(t *testing.T) {
+	tests := []struct {
+		filename string
+		wantDiff *FileDiff
+	}{
+		{
+			filename: "sample_file.diff",
+			wantDiff: &FileDiff{
+				OrigName: "oldname",
+				OrigTime: &pbtypes.Timestamp{Seconds: 1255273940},
+				NewName:  "newname",
+				NewTime:  &pbtypes.Timestamp{Seconds: 1255273950},
+			},
+		},
+		{
+			filename: "sample_file_extended.diff",
+			wantDiff: &FileDiff{
+				OrigName: "oldname",
+				OrigTime: &pbtypes.Timestamp{Seconds: 1255273940},
+				NewName:  "newname",
+				NewTime:  &pbtypes.Timestamp{Seconds: 1255273950},
+				Extended: []string{
+					"diff --git a/vcs/git_cmd.go b/vcs/git_cmd.go",
+					"index aa4de15..7c048ab 100644",
+				},
+			},
+		},
+		{
+			filename: "sample_file_extended_empty_new.diff",
+			wantDiff: &FileDiff{
+				OrigName: "/dev/null",
+				OrigTime: nil,
+				NewName:  "b/vendor/go/build/testdata/empty/dummy",
+				NewTime:  nil,
+				Extended: []string{
+					"diff --git a/vendor/go/build/testdata/empty/dummy b/vendor/go/build/testdata/empty/dummy",
+					"new file mode 100644",
+					"index 0000000..e69de29",
+				},
+			},
+		},
+		{
+			filename: "sample_file_extended_empty_deleted.diff",
+			wantDiff: &FileDiff{
+				OrigName: "a/vendor/go/build/testdata/empty/dummy",
+				OrigTime: nil,
+				NewName:  "/dev/null",
+				NewTime:  nil,
+				Extended: []string{
+					"diff --git a/vendor/go/build/testdata/empty/dummy b/vendor/go/build/testdata/empty/dummy",
+					"deleted file mode 100644",
+					"index e69de29..0000000",
+				},
+			},
+		},
+		{
+			filename: "sample_file_extended_empty_rename.diff",
+			wantDiff: &FileDiff{
+				OrigName: "a/docs/integrations/Email_Notifications.md",
+				OrigTime: nil,
+				NewName:  "b/docs/integrations/email-notifications.md",
+				NewTime:  nil,
+				Extended: []string{
+					"diff --git a/docs/integrations/Email_Notifications.md b/docs/integrations/email-notifications.md",
+					"similarity index 100%",
+					"rename from docs/integrations/Email_Notifications.md",
+					"rename to docs/integrations/email-notifications.md",
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		diffData, err := ioutil.ReadFile(filepath.Join("testdata", test.filename))
+		if err != nil {
+			t.Fatal(err)
+		}
+		diff, err := ParseFileDiff(diffData)
+		if err != nil {
+			t.Fatalf("%s: got ParseFileDiff error %v", test.filename, err)
+		}
+
+		diff.Hunks = nil
+		if got, want := diff, test.wantDiff; !reflect.DeepEqual(got, want) {
+			t.Errorf("%s:\n\ngot: %v\nwant: %v", test.filename, goon.Sdump(got), goon.Sdump(want))
```

Comment:
Could consider using [`github.com/d4l3k/messagediff`](https://github.com/d4l3k/messagediff#example-usage) here instead of printing 2 goons. It might be even more readable.